selectrum.el

```diff
@@ -709,6 +709,9 @@ For PROMPT, DIR, DEFAULT-FILENAME, MUSTMATCH, INITIAL, and
 PREDICATE, see `read-file-name'."
   (let* ((dir (file-name-as-directory
                (expand-file-name (or dir default-directory))))
+         (full-path (if (or initial default-filename)
+                        (expand-file-name (or initial default-filename) dir)
+                      dir))
```

Comment:
I feel like this should be a `when`. Because in the else case, you're setting the default candidate to `dir`, when this wasn't done in the previous code.

Reply:
In the else case, I'm not setting the default candidate to dir. In this case full-path is a directory, and default-candidate becomes an empty string. I didn't notice any difference between using an empty string or nil, but I can change it if you think it should be nil.

Reply:
I don't think there's any difference in practice, because Selectrum doesn't support a candidate that is the empty string, but I think it is much more semantically correct to make it nil.

Reply:
Done.